tests/unit/test_xtcedef.py

```diff
@@ -1762,7 +1762,7 @@ def test_boolean_parameter_type(xml_string, expectation):
             xtcedef.BinaryDataEncoding(fixed_size_in_bits=1)),
          {},
          xtcedef.PacketData(0b0011010000110010010100110000000001001011000000000100100100000000.to_bytes(length=64, byteorder='big')),
-         '0', True),
+         b'', False),
```

Comment:
I think this was actually incorrect before. If I am reading in the 0 bit as a boolean data type I think that should be False. Before we were getting a string representation of the bits which is `"0"`, and `bool("0") == True` because there is something inside the string and it isn't empty.

Reply:
I agree. 

I struggled a bit with truthiness when writing tests. You'll see tests for string encoded data types with values like "false_is_truthy" when encoding a boolean type as a string. Basically unless the value is empty, it became True and I eventually convinced myself that, for strings at least, that's what I wanted. And since I was outputting binary data as strings, the behavior stuck here too.

However, since binary data is basically an int, anything that's all zeros should be falsy. Everything else should be truthy (nonzero). This is a change in behavior but not one that anyone was relying on I'm pretty sure.